apps/src/applab/applab.js

```diff
@@ -262,6 +262,64 @@ function renderFooterInSharedGame() {
   );
 }
 
+Applab.autogenerateML = function(modelId) {
+  $.ajax({
+    url: `/api/v1/ml_models/${modelId}`,
+    method: 'GET'
+  }).then(modelData => {
+    var x = 20;
+    var y = 20;
+    designMode.onInsertEvent(`var testValues = {};`);
+    modelData.selectedFeatures.forEach(feature => {
+      y = y + 20;
+      var label = designMode.createElement('LABEL', x, y);
+      label.textContent = feature + ':';
+      label.id = 'design_' + feature + '_label';
+      label.style.width = '300px';
+      y = y + 20;
+      if (Object.keys(modelData.featureNumberKey).includes(feature)) {
+        var selectId = feature + '_dropdown';
+        var select = designMode.createElement('DROPDOWN', x, y);
+        select.id = 'design_' + selectId;
+        // App Lab automatically addss "option 1" and "option 2", remove them.
+        select.options.remove(0);
+        select.options.remove(0);
+        Object.keys(modelData.featureNumberKey[feature]).forEach(option => {
+          var optionElement = document.createElement('option');
+          optionElement.text = option;
+          select.options.add(optionElement);
+        });
+      } else {
+        var input = designMode.createElement('TEXT_INPUT');
+        input.id = 'design_' + feature + '_input';
+      }
+      var addFeature = `testValues.${feature} = getText("${selectId}");`;
+      designMode.onInsertEvent(addFeature);
+    });
+    y = y + 40;
+    var label = designMode.createElement('LABEL', x, y);
+    label.textContent = modelData.labelColumn;
+    // TODO: this could be problematic if the name isn't formatted appropriately
+    label.id = 'design_' + modelData.name + '_label';
+    label.style.width = '300px';
+    y = y + 20;
+    var predictionId = modelData.name + '_prediction';
+    var prediction = designMode.createElement('TEXT_INPUT', x, y);
+    prediction.id = 'design_' + predictionId;
+    y = y + 40;
+    var predictButton = designMode.createElement('BUTTON', x, y);
+    predictButton.textContent = 'Predict';
+    var predictButtonId = modelData.name + '_predict';
+    designMode.updateProperty(predictButton, 'id', predictButtonId);
+    var predictOnClick = `onEvent("${predictButtonId}", "click", function() {
+      getPrediction("${modelData.name}", testValues, function(value) {
+        setText("${predictionId}", value);
+      });
+    });`;
+    designMode.onInsertEvent(predictOnClick);
+  });
+};
+
```

Comment:
I know `applab.js` is huge already, but wonder whether it might be good to move this into a more AI-related standalone file?